### PR TITLE
AddIsSplitEpimorphism/AddIsSplitMonomorphism

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
I do not understand why the coverage dropped on the lines he shows in the details.

**Comment by zickgraf** [comment]
Since the diff coverage is 100% everything should be fine. The reason why the report is partially red is the reoccurring problem: GAP 4.10 counts covered lines differently than GAP 4.11.

File: gap/FinSetsForCAP.gi [hunk 1]
```diff
@@ -623,6 +627,12 @@ AddIsMonomorphism( FinSets,
     
 end );
 
+##
+AddIsSplitMonomorphism( FinSets,
+  function ( phi )
+    return IsInitial( Range( phi ) ) or ( not IsInitial( Source( phi ) ) and IsMonomorphism( phi ) );
```
Comment (zickgraf) [inline gap/FinSetsForCAP.gi]: `IsInitial` is derived from `UniversalMorphismFromInitialObject` and `IsIsomorphism`. `IsIsomorphism` in turn is derived from `IsMonomorphism` and `IsSplitEpimorphism` but could also be derived from `IsSplitMonomorphism` and `IsEpimorphism`. To make sure we do not run into infinite recursions I propose to implement a primitive version of `IsInitial`.

Comment (zickgraf) [reply gap/FinSetsForCAP.gi]: This is included in the current version.

**Pull request merged**